scripts/sanitize-mocha-results.js

```diff
@@ -0,0 +1,93 @@
+// this is a safety script to ensure that Mocha test reports don't contain any environment variables
+// this script is executed from the verify-mocha-results.js script
+// usage: yarn sanitize:mocha:results
+
+const Bluebird = require('bluebird')
+const fs = require('fs').promises
+const path = require('path')
+const { readCircleEnv } = require('./circle-env')
+
+const REPORTS_PATH = '/tmp/cypress/junit'
+
+// some env is ok in reports. this is based off of what Circle doesn't mask in stdout:
+// https://circleci.com/blog/keep-environment-variables-private-with-secret-masking/
+function isAllowlistedEnv (key, value) {
+  return ['true', 'false', 'TRUE', 'FALSE'].includes(value)
+    || ['nodejs_version', 'CF_DOMAIN', 'SKIP_RELEASE_CHANGELOG_VALIDATION_FOR_BRANCHES', 'CIRCLE_PROJECT_REPONAME', 'HOME', 'PLATFORM', 'HOSTNAME', 'PWD', 'INIT_CWD', 'USER', 'LOGNAME', 'npm_config_loglevel'].includes(key)
+    // ignore npm_package_ envs https://docs.npmjs.com/cli/v11/using-npm/scripts#packagejson-vars
+    || key.startsWith('npm_package_')
+    || value.length < 4
```

Comment:
why less than 4 here? What exactly is this allowing to pass through?

Reply:
The documentation was above the function but I reformatted it [7e76e2e](https://github.com/cypress-io/cypress/pull/31264/commits/7e76e2eeddaebcc20b4fd4fcb9a35224ab1ba520) and [d69c371](https://github.com/cypress-io/cypress/pull/31264/commits/d69c3717d247fb18f9136f520cf9c30d1bfbb120). It's based off what [CircleCI masks](https://circleci.com/docs/env-vars/#secrets-masking). Note, this is not a change from what it was doing before.